src/porepy/utils/adtree.py

```diff
@@ -0,0 +1,438 @@
+                # Intersect with the search box, add in case
+                if self._box_intersect(box, self.nodes[node_id].box):
+                    found.append(node_id)
+                # Go to the left part of the sub-tree, and check if
+                # it intersects the box. The right part is considered later.
+                next_node_id = self.nodes[node_id].child[self.LEFT]
+                if next_node_id != -1:
+                    search_dim = level % self.tree_dim
+                    if search_dim < self.phys_dim:
+                        if origin[search_dim] > box[search_dim + self.phys_dim]:
+                            next_node_id = -1
+                    else:
+                        delta = self._delta(level, self.tree_dim)
+                        if origin[search_dim] + delta < box[search_dim - self.phys_dim]:
+                            next_node_id = -1
+                # Left sub-tree is neither null nor external, push info onto the stack
+                if next_node_id != -1:
+                    stack.append((node_id, origin.copy(), level))
+                    node_id = next_node_id
+                    level += 1
+                else:
+                    break
+
+            # Check the right part that are stored in the stack
+            level += 1
+            node_id = self.nodes[node_id].child[self.RIGHT]
+            while True:
+                while node_id == -1:
+                    if len(stack) == 0:
+                        return np.sort(found)
+                    info = stack.pop()
+                    node_id = self.nodes[info[0]].child[self.RIGHT]
+                    origin = info[1]
+                    level = info[2] + 1
+
+                # Check if the subtree intersects the box. Otherwise set right_link = -1,
+                # pop new node from the stack and adjourn level.
+                #
+                # lev-1 is the level of the parent node, which directs the search.
+                search_dim = (level - 1) % self.tree_dim
+                delta = self._delta(level - 1, self.tree_dim)
+                # reset origin (we have just traversed the tree to the right)
+                origin[search_dim] += delta
+                if search_dim < self.phys_dim:
+                    if origin[search_dim] > box[search_dim + self.phys_dim]:
+                        node_id = -1
+                else:
+                    if origin[search_dim] + delta < box[search_dim - self.phys_dim]:
+                        node_id = -1
+
+                if node_id != -1:
+                    break
+
+        # nothing is found, return an empty array
+        return np.empty(0)
+
+    def from_grid(self, g: pp.Grid, only_cells: Optional[np.ndarray] = None) -> None:
+        """Function that construct the tree from a grid by adding one cells at the time.
+
+        If a portion of the cells should be considered due to some a-priori estimates of the
+        searching elements.
```

Comment:
Isn't this a description of the only_cells parameter?

Reply:
correct, I'll fix it